hoss/dimension_utilities.py

```diff
@@ -55,54 +65,329 @@ def is_index_subset(message: Message) -> bool:
+            projected_dimension_name = 'projected_y'
+        elif override_variable.is_longitude():
+            projected_dimension_name = 'projected_x'
+    return projected_dimension_name
+
+
+def get_override_projected_dimensions(
+    varinfo: VarInfoFromDmr,
+    variable_name: str,
+) -> list[str]:
+    """
+    Returns the projected dimensions names from coordinate variables
+    """
+    coordinate_variables = get_coordinate_variables(varinfo, [variable_name])
+    if coordinate_variables:
+        override_dimensions = []
+        for coordinate in coordinate_variables:
+            override_dimensions.append(
+                get_override_projected_dimension_name(varinfo, coordinate)
+            )
+    else:
+        # if the override is the variable
+        override = get_override_projected_dimension_name(varinfo, variable_name)
+        override_dimensions = ['projected_y', 'projected_x']
+        if (override is None) or (override not in override_dimensions):
+            override_dimensions = []
+    return override_dimensions
+
+
+def get_coordinate_variables(
+    varinfo: VarInfoFromDmr,
+    requested_variables: Set[str],
+) -> list[str]:
+    """This method returns coordinate variables that are referenced
+    in the variables requested. It returns it in a specific order
+    [latitude, longitude]
+    """
+
+    coordinate_variables_set = varinfo.get_references_for_attribute(
+        requested_variables, 'coordinates'
+    )
+    coordinate_variables = []
+    contains_latitude = False
+    contains_longitude = False
+    for coordinate in coordinate_variables_set:
+        if varinfo.get_variable(coordinate).is_latitude():
+            coordinate_variables.insert(0, coordinate)
+            contains_latitude = True
+        elif varinfo.get_variable(coordinate).is_longitude():
+            coordinate_variables.insert(1, coordinate)
+            contains_longitude = True
+
+    if coordinate_variables:
+        if not contains_latitude:
+            raise MissingCoordinateDataset('latitude')
+        if not contains_longitude:
+            raise MissingCoordinateDataset('longitude')
+
+    return coordinate_variables
+
+
+def update_dimension_variables(
+    prefetch_dataset: Dataset, coordinates: Set[str], varinfo: VarInfoFromDmr, crs: CRS
+) -> Dict[str, ndarray]:
+    """Generate artificial 1D dimensions variable for each
+    2D dimension or coordinate variable
+
+    For each dimension variable:
+    (1) Check if the dimension variable is 1D.
+    (2) If it is not 1D and is 2D get the dimension sizes
+    (3) Get the corner points from the coordinate variables
+    (4) Get the x-y max-min values
+    (5) Generate the x-y dimscale array and return to the calling method
+
+    """
+    row_size, col_size = get_row_col_sizes_from_coordinate_datasets(
+        prefetch_dataset, coordinates, varinfo
+    )
+
+    geo_grid_corners = get_geo_grid_corners(prefetch_dataset, coordinates, varinfo)
```

Comment:
This still feels like an unnecessarily complicated methodology. You could just project the entire latitude and longitude arrays using `pyproj` and take one row and one column from the projected output. Is it to reduce the memory usage? SPL3FTP has arrays of (2, 406, 964) and (2, 500, 500) - those size arrays will not be a problem for memory usage.

The method we have here adds a _bunch_ of functions and complexity that would otherwise not be needed.

Reply:
will pyproj handle cases with fill values?